### DISCUSSION TITLE
Avoid dylib from linkage fixing

### DISCUSSION THREAD

**Discussion by daviderestivo** [Homebrew/discussions#274] [Tap maintenance and Homebrew development]
Hi,

I'm currently maintaining [a brew formula for GNU/Emacs](https://github.com/daviderestivo/homebrew-emacs-head). Since some time a new branch of Emacs is being develped allowing LISP code compilation into native code (dylib). More details here: http://akrl.sdf.org/gccemacs.html

I'm trying to modify my formula in order to support this new feature (native-comp branch) of GNU/Emacs HEAD. The installation is successful but it seems that after the "install" phase homebrew changes the DYLIB ID of the native compiled Elisp code breaking someway those libraries:

```
==> Changing dylib ID of /usr/local/Cellar/emacs-head@28/28.0.50_1/Emacs.app/Contents/native-lisp/28.0.50-x86_64-apple-darwin19.6.0-280c4c724fff373c5b7192112ad4147c/PY-b5-9da27250173071085086a7aed29fd29d-7a2e0ead8fc81e02717b53d465b7c866.eln 
  from /private/tmp/libgccjit-3EU2CE/fake.so 
    to /usr/local/opt/emacs-head@28/Emacs.app/Contents/native-lisp/28.0.50-x86_64-apple-darwin19.6.0-280c4c724fff373c5b7192112ad4147c/fake.so 
==> Changing dylib ID of /usr/local/Cellar/emacs-head@28/28.0.50_1/Emacs.app/Contents/native-lisp/28.0.50-x86_64-apple-darwin19.6.0-280c4c724fff373c5b7192112ad4147c/PY-df2668c28394ecb13448539e114796c3-49c85c7a1e624e4fcb832e0ae1789cce.eln 
  from /private/tmp/libgccjit-MdVbDk/fake.so 
    to /usr/local/opt/emacs-head@28/Emacs.app/Contents/native-lisp/28.0.50-x86_64-apple-darwin19.6.0-280c4c724fff373c5b7192112ad4147c/fake.so 
==> Changing dylib ID of /usr/local/Cellar/emacs-head@28/28.0.50_1/Emacs.app/Contents/native-lisp/28.0.50-x86_64-apple-darwin19.6.0-280c4c724fff373c5b7192112ad4147c/Punct-9d38e5d98c73f73af49bda4ce95ce922-b334c1a512d9d6beda163f5975881cb6.eln 
  from /private/tmp/libgccjit-sjtPHL/fake.so 
```

I would like to know if there is a way to avoid this behaviour.

Thanks in advance and regards,
Davide

**Comment by SMillerDev** [comment]
I think the best way forward is to report this bug to homebrew/brew so it can be fixed instead of turning it off.

**Reply by daviderestivo** [reply]
@SMillerDev, sure I can do it ... thanks for the quick answer :)

**Reply by daviderestivo** [reply]
Done ... thanks.

Ref: https://github.com/Homebrew/homebrew-core/issues/66810

**Reply by SMillerDev** [reply]
That's homebrew/homebrew-core, which contains just the software that's shipped. Not the actual brew tool. That one is homebrew/brew.

**Reply by daviderestivo** [reply]
Ok then I'll  move it to the brew tool ... thanks. 

EDIT: someone already did it...
